src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -53,7 +56,8 @@ internal class DefaultRazorLanguageServerCustomMessageTarget : RazorLanguageServ
         LSPRequestInvoker requestInvoker,
         FormattingOptionsProvider formattingOptionsProvider,
         EditorSettingsManager editorSettingsManager,
-        LSPDocumentSynchronizer documentSynchronizer)
+        LSPDocumentSynchronizer documentSynchronizer,
+        OutputWindowLogger logger)
```

Comment:
This still seems wrong that its not ILogger, or LoggerFactory, or whatever. Is there a reason?

Reply:
If `DefaultRazorLanguageSersverCustomMessageTarget` was constructed using our own DI I'd agree, but because it uses MEF I am hesitant to export `OutputWindowLogger` as an `ILogger` because then (AFAIK) it would become available to anyone doing `[ImportMany]IEnumerable<ILogger> loggers` within the whole of VS, which seems like a bad time.

If there is some reason why that is not the case or way to avoid it I'm all ears.

Reply:
Ahh, okay, that makes sense. We could create a new interface and MEF export the loggers as that specifically? I'm just a bit worried that someone will add calls to `_logger.Log(..)` in this class, and then spend hours trying to debug why they're not in any log files :)